File: integrations/aspnetcore/src/Scalar.AspNetCore/Options/Authentication/Schemes/ScalarSecurityScheme.cs
using System.Text.Json.Serialization;

namespace Scalar.AspNetCore;

/// <summary>
/// Represents the base class for security schemes in Scalar.
/// </summary>
[JsonDerivedType(typeof(ScalarHttpSecurityScheme))]
[JsonDerivedType(typeof(ScalarApiKeySecurityScheme))]
[JsonDerivedType(typeof(ScalarOAuth2SecurityScheme))]
public abstract class ScalarSecurityScheme
{
    /// <summary>
    /// Gets or sets the description of this security scheme.
    /// </summary>
    public string? Description { get; set; }
}

File: integrations/java/webjar/src/main/java/com/scalar/maven/webjar/authentication/schemes/ScalarSecurityScheme.java
package com.scalar.maven.webjar.authentication.schemes;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

/**
 * Represents the base class for security schemes in Scalar.
 */
@JsonTypeInfo(use = JsonTypeInfo.Id.DEDUCTION)
@JsonSubTypes({
        @JsonSubTypes.Type(value = ScalarHttpSecurityScheme.class),
        @JsonSubTypes.Type(value = ScalarApiKeySecurityScheme.class),
        @JsonSubTypes.Type(value = ScalarOAuth2SecurityScheme.class)
})
public abstract class ScalarSecurityScheme {

    /**
     * Gets or sets the description of this security scheme.
     */
    private String description;

    /**
     * Creates a new ScalarSecurityScheme.
     */
    protected ScalarSecurityScheme() {
    }

    /**
     * Gets the description of this security scheme.
     *
     * @return the description
     */
    public String getDescription() {
        return description;
    }

    /**
     * Sets the description of this security scheme.
     *
     * @param description the description
     */
    public void setDescription(String description) {
        this.description = description;
    }
}
